egs/ami/s5/local/ami_format_data.sh

```diff
@@ -15,25 +15,12 @@ arpa_lm=$1
 
 cp -r data/lang data/lang_test
 
-# grep -v '<s> <s>' etc. is only for future-proofing this script.  Our
-# LM doesn't have these "invalid combinations".  These can cause 
-# determinization failures of CLG [ends up being epsilon cycles].
-# Note: remove_oovs.pl takes a list of words in the LM that aren't in
-# our word list.  Since our LM doesn't have any, we just give it
-# /dev/null [we leave it in the script to show how you'd do it].
 gunzip -c "$arpa_lm" | \
-   grep -v '<s> <s>' | \
-   grep -v '</s> <s>' | \
-   grep -v '</s> </s>' | \
-   arpa2fst - | fstprint | \
-   utils/remove_oovs.pl /dev/null | \
-   utils/eps2disambig.pl | utils/s2eps.pl | fstcompile --isymbols=data/lang_test/words.txt \
-     --osymbols=data/lang_test/words.txt  --keep_isymbols=false --keep_osymbols=false | \
-    fstrmepsilon | fstarcsort --sort_type=ilabel > data/lang_test/G.fst
-  fstisstochastic data/lang_test/G.fst
+  arpa2fst --disambig-symbol=#0 \
```

Comment:
I just don't like the look of having the explicit "-" followed by a
redirection- it seems redundant.  I.e. if you're going write - and then
write the filename, you could just write the filename.  I'd put the
filename there directly.

On Fri, Apr 1, 2016 at 1:39 AM, Kirill Katsnelson <notifications@github.com>
wrote:

> In egs/ami/s5/local/ami_format_data.sh
> <https://github.com/kaldi-asr/kaldi/pull/634#discussion_r58164126>:
>
> >  gunzip -c "$arpa_lm" | \
> > -   grep -v '<s> <s>' | \
> > -   grep -v '</s> <s>' | \
> > -   grep -v '</s> </s>' | \
> > -   arpa2fst - | fstprint | \
> > -   utils/remove_oovs.pl /dev/null | \
> > -   utils/eps2disambig.pl | utils/s2eps.pl | fstcompile --isymbols=data/lang_test/words.txt \
> > -     --osymbols=data/lang_test/words.txt  --keep_isymbols=false --keep_osymbols=false | \
> > -    fstrmepsilon | fstarcsort --sort_type=ilabel > data/lang_test/G.fst
> > -  fstisstochastic data/lang_test/G.fst
> > +  arpa2fst --disambig-symbol=#0 \
>
> Rather ... | arpa2fst - data/lang_test/G.fst -- is that what you mean? I
> can do that as well, just a matter of style. It looks to me most pipelines
> in the scripts end in a redirection. But I am totally ambivalent about
> these, which way would you prefer?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634/files/7f89580915a7568b2fa897570b041cfcb2e96108#r58164126>
>